setup.py

```diff
@@ -33,8 +33,12 @@ def _import(module, function=None, alias=None):
     try:
         _import(**dep)
     except ImportError:
-        import pip
-        pip.main(['install', lib])
+        try:
+            from pip import main # old API
+        except ImportError:
+            from pip._internal import main # new API
```

Comment:
I’m surprised that the new recommended approach would involve importing from a non-public internal module. Any idea if there is a way to do this using public APIs?

Reply:
There is no officially supported public API for pip. 

https://groups.google.com/forum/#!topic/pypa-dev/JVTfS6ZdAuM

The alternative is `subprocess.call`.

https://stackoverflow.com/a/50255019